pkg/operation/botanist/secrets.go

```diff
@@ -724,6 +731,29 @@ func (b *Botanist) DeploySecrets(ctx context.Context) error {
 		}
 	}
 
+	// The kubeconfig secrets for the dependency-watchdog probes have now stopped using token to avoid the need for reconciliation
+	// on rotation. The old secrets containing tokens must be removed so that new secrets without token can be generated.
+	for _, secretName := range []string{common.DependencyWatchdogInternalProbeSecretName, common.DependencyWatchdogExternalProbeSecretName} {
+		secret := &corev1.Secret{}
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, secretName), secret); err != nil {
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			return err
+		}
+		if restconfig, err := kubernetes.NewRESTConfigFromSecret(secret); err != nil {
```

Comment:
Hm, can we save some code here and just check if the `token` key exists in `secret.Data`? There is no need to create a REST config. If `token` key exists it is enough to know that the secret shall be deleted.

Reply:
Done